src/components/structures/GroupView.js

```diff
@@ -726,7 +727,8 @@ export default createReactClass({
 
     _onJoinClick: async function() {
         if (this._matrixClient.isGuest()) {
-            dis.dispatch({action: 'require_registration'});
+            dis.dispatch({action: 'require_registration',
+                          screen_after: {screen: `group/${this.props.groupId}`}});
```

Comment:
I couldn't see a way of fulfilling the join here, you end up clicking the button twice, which isn't the end of the world but is still annoying.